From 0d45769add0f5f93ab0c20d38432c60090bda853 Mon Sep 17 00:00:00 2001
From: Quanyang Wang <quanyang.wang@windriver.com>
Date: Thu, 4 Apr 2019 17:23:32 +0800
Subject: [PATCH 12/30] drm/imx/dcss: fix compile error in dcss-kms.c

In upstream commit 77ac3b00b131 ("drm/atomic: Remove deprecated accessor macros"),
for_each_connector_in_state has been removed and should use for_each_new_connector_in_state
instead.

Signed-off-by: Quanyang Wang <quanyang.wang@windriver.com>
---
 drivers/gpu/drm/imx/dcss/dcss-kms.c | 2 +-
 drivers/gpu/drm/imx/dpu/dpu-kms.c   | 6 +++---
 drivers/gpu/drm/mxsfb/mxsfb_drv.c   | 2 +-
 3 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/drivers/gpu/drm/imx/dcss/dcss-kms.c b/drivers/gpu/drm/imx/dcss/dcss-kms.c
index 076a3cb2da4d..ef00e81d5647 100644
--- a/drivers/gpu/drm/imx/dcss/dcss-kms.c
+++ b/drivers/gpu/drm/imx/dcss/dcss-kms.c
@@ -54,7 +54,7 @@ static void dcss_kms_setup_output_pipe(struct drm_atomic_state *state)
 	struct drm_display_info *di;
 	int i;
 
-	for_each_connector_in_state(state, connector, conn_state, i) {
+	for_each_new_connector_in_state(state, connector, conn_state, i) {
 		if (!connector->state->best_encoder)
 			continue;
 
diff --git a/drivers/gpu/drm/imx/dpu/dpu-kms.c b/drivers/gpu/drm/imx/dpu/dpu-kms.c
index 0c674dc8445a..1709fed6308e 100644
--- a/drivers/gpu/drm/imx/dpu/dpu-kms.c
+++ b/drivers/gpu/drm/imx/dpu/dpu-kms.c
@@ -411,7 +411,7 @@ dpu_atomic_mark_pipe_states_prone_to_put_per_crtc(struct drm_crtc *crtc,
 	int i;
 
 	if ((crtc_mask & drm_crtc_mask(crtc)) == 0) {
-		for_each_plane_in_state(state, plane, plane_state, i) {
+		for_each_new_plane_in_state(state, plane, plane_state, i) {
 			if (plane->possible_crtcs &
 			    drm_crtc_mask(crtc)) {
 				found_pstate = true;
@@ -552,7 +552,7 @@ static int dpu_drm_atomic_check(struct drm_device *dev,
 		grp[i] = NULL;
 	}
 
-	for_each_crtc_in_state(state, crtc, crtc_state, i)
+	for_each_new_crtc_in_state(state, crtc, crtc_state, i)
 		crtc_mask_in_state |= drm_crtc_mask(crtc);
 
 	drm_for_each_crtc(crtc, dev) {
@@ -710,7 +710,7 @@ static int dpu_drm_atomic_check(struct drm_device *dev,
 	if (ret)
 		return ret;
 
-	for_each_crtc_in_state(state, crtc, crtc_state, i) {
+	for_each_new_crtc_in_state(state, crtc, crtc_state, i) {
 		struct dpu_crtc *dpu_crtc = to_dpu_crtc(crtc);
 		struct drm_plane_state **states;
 		int n;
diff --git a/drivers/gpu/drm/mxsfb/mxsfb_drv.c b/drivers/gpu/drm/mxsfb/mxsfb_drv.c
index 39f7bb487409..4407307fd19e 100644
--- a/drivers/gpu/drm/mxsfb/mxsfb_drv.c
+++ b/drivers/gpu/drm/mxsfb/mxsfb_drv.c
@@ -127,7 +127,7 @@ static int mxsfb_atomic_helper_check(struct drm_device *dev,
 	if (ret)
 		return ret;
 
-	for_each_crtc_in_state(state, crtc, crtc_state, i) {
+	for_each_new_crtc_in_state(state, crtc, crtc_state, i) {
 		struct drm_plane_state *primary_state;
 		int old_bpp = 0;
 		int new_bpp = 0;
-- 
2.17.1

